examples/pkg/kcp/clients/informers/example/v1/clustertesttype.go

```diff
@@ -0,0 +1,100 @@
+//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
+
+/*
+Copyright The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by kcp code-generator. DO NOT EDIT.
+
+package v1
+
+import (
+	"context"
+	"time"
+
+	kcpcache "github.com/kcp-dev/apimachinery/pkg/cache"
+	kcpinformers "github.com/kcp-dev/apimachinery/third_party/informers"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+
+	examplev1 "acme.corp/pkg/apis/example/v1"
+	clientset "acme.corp/pkg/kcp/clients/clientset/versioned"
+	"acme.corp/pkg/kcp/clients/informers/internalinterfaces"
+	examplev1listers "acme.corp/pkg/kcp/clients/listers/example/v1"
+)
+
+// ClusterTestTypeClusterInformer provides access to a shared informer and lister for
+// ClusterTestTypes.
+type ClusterTestTypeClusterInformer interface {
+	Informer() cache.SharedIndexInformer
+	Lister() examplev1listers.ClusterTestTypeClusterLister
+}
+
+type clusterTestTypeClusterInformer struct {
+	factory          internalinterfaces.SharedInformerFactory
+	tweakListOptions internalinterfaces.TweakListOptionsFunc
+}
+
+// NewClusterTestTypeClusterInformer constructs a new informer for ClusterTestType type.
+// Always prefer using an informer factory to get a shared informer instead of getting an independent
+// one. This reduces memory footprint and number of connections to the server.
+func NewClusterTestTypeClusterInformer(client clientset.ClusterInterface, resyncPeriod time.Duration, indexers cache.Indexers) cache.SharedIndexInformer {
```

Comment:
definite todo: collapse to only 1 constructor (i.e. don't mirror upstream), `client` is only required param, everything else is functional options.